### PR TITLE
uf-8347: Email batch implemented.

### PR DESCRIPTION
**Pull request by mha109**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -11,15 +11,15 @@
     </parent>
     <groupId>se.sundsvall.messaging</groupId>
     <artifactId>api-service-messaging</artifactId>
-    <version>4.4</version>
+    <version>4.5</version>
```
Comment (jklingst) [inline pom.xml:14]: New and shiny 5.0.1 ?

File: src/main/java/se/sundsvall/messaging/api/model/request/EmailBatchRequest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import lombok.Builder;
+import lombok.With;
+
+@With
+@Builder(setterPrefix = "with")
+public record EmailBatchRequest(
+
+	@Valid
+	@ArraySchema(arraySchema = @Schema(implementation = Party.class))
+	List<Party> parties,
```
Comment (jklingst) [inline src/main/java/se/sundsvall/messaging/api/model/request/EmailBatchRequest.java:28]: Dunno if it's checked manually elsewhere, but maybe:
```
@NotEmpty
@ArraySchema(...)
List<@Valid Party> parties,
```

File: src/main/java/se/sundsvall/messaging/service/mapper/RequestMapper.java [hunk 3]
```diff
@@ -20,128 +21,128 @@
...
+			.withSender(sender)
+			.withMobileNumber(mobileNumber)
+			.withMessage(originalMessage.message())
+			.withOrigin(message.origin())
+			.build();
+
+		return toJson(smsRequest);
+	}
+
+	public String toEmailBatchRequest(final Message message, final String emailAddress) {
```
Comment (jklingst) [inline src/main/java/se/sundsvall/messaging/service/mapper/RequestMapper.java:64]: Should this method really have been renamed ?

File: src/main/java/se/sundsvall/messaging/service/MessageEventDispatcher.java [hunk 4]
```diff
@@ -79,14 +81,22 @@ public InternalDeliveryResult handleEmailRequest(final EmailRequest request) {
...
+		var batchId = UUID.randomUUID().toString();
 
-        var message = dbIntegration.saveMessage(messageMapper.toMessage(request));
+		var deliveryResults = ofNullable(request.parties()).orElse(Collections.emptyList()).stream()
+			.filter(party -> isWhitelisted(MessageType.EMAIL, party.emailAddress()))
+			.map(party -> requestMapper.toEmailBatchRequest(request, party))
+			.map(emailRequest -> messageMapper.toMessage(emailRequest, batchId))
+			.map(dbIntegration::saveMessage)
+			.map(this::publishMessageEvent)
+			.toList();
```
Comment (AndreSoderlund) [inline src/main/java/se/sundsvall/messaging/service/MessageEventDispatcher.java:93]: Could possibly map directly from party -> message.

File: src/main/java/se/sundsvall/messaging/service/mapper/RequestMapper.java (hunk 3 above)
Comment (mha109) [reply src/main/java/se/sundsvall/messaging/service/mapper/RequestMapper.java:64]: Nope

File: src/main/java/se/sundsvall/messaging/service/MessageEventDispatcher.java (hunk 4 above)
Comment (mha109) [reply src/main/java/se/sundsvall/messaging/service/MessageEventDispatcher.java:93]: If it doesn't upset you I'l leave it "as is" for clarity :)

File: src/main/java/se/sundsvall/messaging/service/BlacklistService.java [hunk 5]
```diff
@@ -55,9 +55,13 @@ public void check(final EmailRequest request) {
         check(EMAIL, request.emailAddress());
     }
 
-    public void check(final DigitalMailRequest request) {
-        request.party().partyIds().forEach(partyId -> check(DIGITAL_MAIL, partyId));
-    }
+	public void check(final String emailAddress) {
```
Comment (josc04) [inline src/main/java/se/sundsvall/messaging/service/BlacklistService.java:58]: This isn't used anymore and can be removed, right?

File: src/test/java/se/sundsvall/messaging/api/MessageResourceFailureTests.java
Comment (josc04) [inline src/test/java/se/sundsvall/messaging/api/MessageResourceFailureTests.java:1]: Either keep this class and remove EmailBatchRequestConstraintValidationTests.java or vice versa, they do the same thing. Personally I would keep this class, extend it with other validation failure tests and remove all *ConstraintValidationTests - but that is probably an own task. For now, maybe just remove EmailBatchRequestConstraintValidationTests and add a task to move other constraintValidationTests into this class?

Comment (mha109) [reply src/test/java/se/sundsvall/messaging/api/MessageResourceFailureTests.java:1]: I'll remove it and make a new task.

**Review by AndreSoderlund — APPROVED** [review]

**Pull request merged**